pkg/build/api/types.go

```diff
@@ -206,11 +209,22 @@ type STIBuildStrategy struct {
 // BuildOutput is input to a build strategy and describes the Docker image that the strategy
 // should produce.
 type BuildOutput struct {
-	// ImageTag is the tag to give to the image resulting from the build.
-	ImageTag string `json:"imageTag,omitempty" yaml:"imageTag,omitempty"`
+	// To defines an optional ImageRepository to push the output of this build to. The namespace
+	// may be empty, in which case the named ImageRepository will be retrieved from the namespace
+	// of the build. Kind must be set to 'ImageRepository' and is the only supported value. If set,
+	// this field takes priority over DockerImageReference. This value will be used to look up
+	// a Docker image repository to push to.
+	To *kapi.ObjectReference `json:"to,omitempty" yaml:"to,omitempty"`
+
+	// Tag is the "version name" that will be associated with the output image. This
+	// field is only used if the To field is set, and is ignored when DockerImageReference is used.
+	// This value represents a consistent name for a set of related changes (v1, 5.x, 5.5, dev, stable)
+	// and defaults to the preferred tag for "To" if not specified.
+	Tag string `json:"tag,omitempty" yaml:"tag,omitempty"`
 
-	// Registry is the Docker registry which should receive the resulting built image via push.
-	Registry string `json:"registry,omitempty" yaml:"registry,omitempty"`
+	// DockerImageReference is the full name of an image ([registry/]name[:tag]), and will be the
+	// value sent to Docker push at the end of a build.
```

Comment:
"will be the value sent to Docker push at the end of the build if there is no To ImageRepo reference defined"

should also clarify in the comments how invalid/missing ImageRepo references are handled (fallback to the DockerImageReference value, fail the build, or do nothing?)